research/2022-07-15-relay-anonymity.md

```diff
@@ -0,0 +1,351 @@
+
+---
+layout: post
+name:  "Waku Privacy and Anonymity Analysis Part I: Definitions and Waku Relay"
+title: "Waku Privacy and Anonymity Analysis Part I: Definitions and Waku Relay"
+date: 2022-07-15 10:00:00
+author: Daniel
+published: true
+permalink: /wakuv2-relay-anon
+categories: research
+summary: Introducing a basic threat model and privacy/anonymity analysis for the Waku v2 relay protocol.
+image: /assets/img/vac.png
+discuss: https://forum.vac.dev/t/discussion-waku-privacy-and-anonymity-analysis/149
+---
+
+<!-- intro -->
+
+[Waku v2](https://rfc.vac.dev/spec/10/) enables secure, privacy preserving communication using a set of modular P2P protocols.
+Waku v2 also aims at protecting the user's anonymity.
+This post is the first in a series about Waku v2 security, privacy, and anonymity.
+The goal is to eventually have a full privacy and anonymity analysis for each of the Waku v2 protocols, as well as covering the interactions of various Waku v2 protocols.
+This provides transparency with respect to Waku's current privacy and anonymity guarantees, and also identifies weak points that we have to address.
+
+In this post, we first give an informal description of security, privacy and anonymity in the context of Waku v2.
+For each definition, we summarize Waku's current guarantees regarding the respective property.
+We also provide attacker models, an attack-based threat model, and a first anonymity analysis of [Waku v2 relay](https://rfc.vac.dev/spec/11/) within the respective models.
+
+Waku comprises many protocols that can be combined in a modular way.
+For our privacy and anonymity analysis, we start with the relay protocol because it is at the core of Waku v2 enabling Waku's publish subscribe approach to P2P messaging.
+In its current form, Waku relay is a minor extension of [libp2p GossipSub](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/README.md).
+
+## Informal Definitions: Security, Privacy, and Anonymity
+
+The concepts of security, privacy, and anonymity are linked and have quite a bit of overlap.
+
+### Security
+
+Of the three, [Security](https://en.wikipedia.org/wiki/Information_security) has the clearest agreed upon definition,
+at least regarding its key concepts: *confidentiality*, *integrity*, and *availability*.
+
+* confidentiality:  data is not disclosed to unauthorized entities.
+* integrity: data is not modified by unauthorized entities.
+* availability: data is available, i.e. accessible by authorized entities.
```

Comment:
Do we want to mention "censorship resistance" as a subsection here somehow? It is definitely a security goal in the broader sense, and might fit in under "availability"

Reply:
It is related, but I'd leave this section short and just cover the three key aspects of security.
I mentioned censorship resitstance and pluggable transport in the post, but it is out of focus for now.
I could add a short censorship resistance section following the anonymity section to show more prominently that it is on our agenda.